frontend/src/containers/EditLogo.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from "react";
+import { useHistory } from "react-router-dom";
+import { useForm } from "react-hook-form";
+import { useSettings, useLogo } from "../hooks";
+import BackendService from "../services/BackendService";
+import ContentService from "../services/ContentService";
+import FileInput from "../components/FileInput";
+import Button from "../components/Button";
+import Breadcrumbs from "../components/Breadcrumbs";
+import defaultLogo from "../logo.svg";
+
+function EditLogo() {
+  const history = useHistory();
+  const { settings, reloadSettings } = useSettings(true);
+  const { logo } = useLogo(settings.customLogoS3Key);
+  const { register, handleSubmit } = useForm();
+
+  const [currentLogo, setCurrentLogo] = useState(logo);
+  const [imageUploading, setImageUploading] = useState(false);
+
+  const onSubmit = async () => {
+    if (currentLogo) {
+      try {
+        setImageUploading(true);
+
+        const s3Key = await ContentService.uploadLogo(currentLogo);
+
+        await BackendService.updateSetting(
+          "customLogoS3Key",
+          s3Key,
+          new Date()
+        );
+        await reloadSettings();
+
+        setImageUploading(false);
+
+        history.push("/admin/settings/brandingandstyling", {
+          alert: {
+            type: "success",
+            message: "Logo successfully modified",
+          },
+        });
+      } catch (err) {
+        console.log("Failed to save logo", err);
```

Comment:
Show an error alert if this happens instead of console log?
```js
history.push("/admin/settings/brandingandstyling", {
          alert: {
            type: "error",
            message: "Failed to update logo",
          },
});
```

Reply:
Good call.